Code/Framework/AzCore/AzCore/Settings/SettingsRegistryMergeUtils.cpp

```diff
@@ -36,34 +38,186 @@ namespace AZ::Internal
+        {
+            projectEngineMoniker = engineDependency.GetName();
+        }
+
+        if (projectEngineMoniker != engineMoniker)
+        {
+            return AZ::Failure(AZStd::string::format("Project engine name '%s' does not match engine name '%s'",
+                projectEngineMoniker.c_str(), engineMoniker.c_str()));
+        }
+
+        if (engineDependency.GetBounds().empty())
+        {
+            // There is no version specifier to satisfy
+            return AZ::Success();
+        }
+
+        // If the engine has no version information or is not known incompatible then assume compatible 
+        if(FixedValueString engineVersionValue;
+           settingsRegistry.Get(engineVersionValue, FixedValueString::format("%s/version", EngineSettingsRootKey)))
+        {
+            AZ::SemanticVersion engineVersion;
+            engineVersion.ParseFromString(engineVersionValue.c_str());
+            if(!engineDependency.IsFullfilledBy(AZ::Specifier<AZ::SemanticVersion::parts_count>(AZ::Uuid::CreateNull(), engineVersion)))
+            {
+                return AZ::Failure(AZStd::string::format("Engine version '%s' does not satisfy the project's engine constraints '%s' ",
+                    engineVersionValue.c_str(), projectEngineMonikerWithSpecifier.c_str()));
+            }
+        }
+
+        return AZ::Success();
+    }
+
+    AZ::IO::FixedMaxPath ReconcileEngineRootFromProjectUserPath(
+        SettingsRegistryInterface& settingsRegistry, const AZ::IO::FixedMaxPath& projectPath)
+    {
+        using namespace AZ::SettingsRegistryMergeUtils;
+        using FixedValueString = AZ::SettingsRegistryInterface::FixedValueString;
+        constexpr auto format = AZ::SettingsRegistryInterface::Format::JsonMergePatch;
+
+        AZ::IO::FixedMaxPath projectUserPath;
+        if (!settingsRegistry.Get(projectUserPath.Native(), FilePathKey_ProjectUserPath) || projectUserPath.empty())
+        {
+            return {};
+        }
+
+        if (auto outcome = MergeSettingsFile(settingsRegistry, (projectUserPath / Internal::ProjectJsonFilename).LexicallyNormal(), ProjectSettingsRootKey);
+            !outcome)
+        {
+            return {};
+        }
+
+        AZ::IO::FixedMaxPath engineRoot;
+        settingsRegistry.Get(engineRoot.Native(), FixedValueString::format("%s/engine_path", ProjectSettingsRootKey));
+        if (!engineRoot.empty())
+        {
+            if (engineRoot.IsRelative())
+            {
+                if (auto engineRootAbsPath = AZ::Utils::ConvertToAbsolutePath(engineRoot.Native());
+                    engineRootAbsPath.has_value())
+                {
+                    engineRoot = AZStd::move(*engineRootAbsPath);
+                }
+            }
+
+            AZ::IO::FixedMaxPath engineJsonPath{ engineRoot / Internal::EngineJsonFilename };
+            AZ::IO::FixedMaxPath projectJsonPath{ engineRoot / projectPath / Internal::ProjectJsonFilename };
```

Comment:
The projectPath is absolute at this point, so it appending it to the engineRoot is a no op
```suggestion
            AZ::IO::FixedMaxPath projectJsonPath{ projectPath / Internal::ProjectJsonFilename };
```